dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/EnhancedResultsFigureGenerator.java

```diff
@@ -104,16 +125,22 @@ private static String generate18(String projectName, double[] percentages, doubl
     /*
      * a private method that generates Figure 19
      */
-    private static String generate19(String projectName, double[] orig_values, double[] time_values, double orig_value,
-            List<GeometricMeanData> fig19GeoData) {
+    private static String generate19(String projectName, double[] orig_values, double[] time_values,
+            double orig_time_value, List<GeometricMeanData> fig19GeoData, boolean[] nonZero, double orig_apfd_value) {
         String result = projectName;
         double calc1 = 0;
         double calc2 = 0;
         double calc3 = 0;
         for (int i = 0; i + 2 <= orig_values.length; i += 2) {
-            calc1 = (orig_values[i] / orig_value);
-            calc2 = (orig_values[i + 1] / orig_value);
-            calc3 = calc2 - calc1;
+            if (!nonZero[i]) {
+                calc1 = (orig_values[i] / orig_time_value);
+                calc2 = (orig_values[i + 1] / orig_time_value);
+                calc3 = calc2 - calc1;
+            } else {
+                calc1 = (orig_values[i] / orig_time_value);
```

Comment:
Another suggestion would be instead of removing them is to add support for a debug flag. When the debug flag is passed then these statements would print otherwise they will not. You can decide whether you want to remove them or add a debug flag.